Sources/TUSKit/TUSClient.swift

```diff
@@ -203,26 +204,28 @@ public final class TUSClient: NSObject {
         do {
             let id = UUID()
             
-            func makeMetadata() throws -> UploadMetadata {
-                guard let files = self.files else {
-                    throw TUSClientError.couldNotUploadFile
+            try autoreleasepool {
```

Comment:
all the things inside of an autoreleasepool need to return an autoreleasepool in order for it to do what it is supposed to do.
https://swiftrocks.com/autoreleasepool-in-swift

Reply:
Hmm, I'll run some more tests to compare. But when I added this I noticed during initial copying to cache the memory only spiked like 30% of what it normally does.

Reply:
Here's no autoreleasepool 

https://user-images.githubusercontent.com/92157216/183923212-457d83df-ecfb-4079-b8ab-31207df9a14e.mov

Reply:
Here's with autoreleasepool

https://user-images.githubusercontent.com/92157216/183923308-fa8ca76e-b1f1-4fc1-b19e-b063e0eeee12.mov